rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEjbConfigurationXmlRuleProvider.java

```diff
@@ -509,27 +512,48 @@ private void processEntityElement(GraphContext ctx, EjbDeploymentDescriptorModel
         return resources;
     }
 
-    private JavaClassModel getOrCreatePhantom(JavaClassService service, String fqcn)
+    private JavaClassModel getOrCreatePhantom(GraphRewrite event, EvaluationContext context, JavaClassService service, String fqcn)
     {
         JavaClassModel classModel = service.getOrCreatePhantom(fqcn);
         if (classModel instanceof AmbiguousJavaClassModel)
         {
             for (JavaClassModel reference : ((AmbiguousJavaClassModel) classModel).getReferences())
             {
-                markAsReportReportable(reference);
+                markAsReportReportable(event, context, reference);
             }
         }
         else if (!(classModel instanceof PhantomJavaClassModel))
         {
-            markAsReportReportable(classModel);
+            markAsReportReportable(event, context, classModel);
         }
         return classModel;
     }
 
-    private void markAsReportReportable(JavaClassModel reference)
+    private void markAsReportReportable(GraphRewrite event, EvaluationContext context, JavaClassModel reference)
     {
         JavaSourceFileModel originalSource = reference.getOriginalSource();
         JavaSourceFileModel decompiledSource = reference.getDecompiledSource();
+        if (originalSource == null && decompiledSource == null && reference.getClassFile() != null
+                    && reference.getClassFile() instanceof JavaClassFileModel)
+        {
+            JavaClassFileModel javaClassFileModel = (JavaClassFileModel) reference.getClassFile();
+            javaClassFileModel.setSkipDecompilation(false);
+
+            // try to decompile it
+            FernflowerDecompilerOperation decompilerOperation = new FernflowerDecompilerOperation();
```

Comment:
We often discover EJBs that are not interesting in and of themselves. They are only interesting because they are EJBs. The decompilation filter would have skipped them, though, so we have to go back and decompile them lazily.